self-core-impl/src/main/java/com/selfxdsd/core/GithubStars.java

```diff
@@ -78,12 +81,99 @@ public boolean add() {
         } else {
             starred = false;
             LOG.error(
-                "Unexpected status when starring repo " + repoName
-                + " of user " + owner + " ]"
-                + "Expected 204 NO CONTENT, but got "
-                + response.statusCode()
+                "Unexpected status when starring repo "
+                    + this.holder.getRepo()
+                    + " of user " + this.holder.getOwner() + " ]"
+                    + "Expected 204 NO CONTENT, but got "
+                    + response.statusCode()
             );
         }
         return starred;
     }
+
+    /**
+     * {@inheritDoc}
+     * <br/>
+     * See doc <a href="https://docs.github.com/en/rest/reference/activity#check-if-a-repository-is-starred-by-the-authenticated-user">here</a>.
+     */
+    @Override
+    public boolean isStarred() {
+        LOG.debug(
+            "Check if Github repository " + this.holder.getRepo()
+                + " is starred by current authenticated user."
+        );
+
+        final boolean isStarred;
+        final Resource response = this.resources.get(
+            URI.create(this.starsUri.toString()),
+            () -> Map.of("Accept", List.of("application/vnd.github.v3+json"))
+        );
+        if (response.statusCode() == HttpURLConnection.HTTP_NO_CONTENT) {
```

Comment:
@criske according to the docs: https://docs.github.com/en/rest/reference/activity#check-if-a-repository-is-starred-by-the-authenticated-user

This endpoint can also return 304 NOT MODIFIED. This means, the request will go through our ConditionalJsonResources mechanism and it will come out as 200 OK and it will break this logic. 

Can you also add ``|| statusCode == OK``? :D

Reply:
@amihaiemil You can get a 304 only if your request headers has an ["If-None-Match" : <ETag>] entry.

I think the only way to get a 304 is that a prior response from a post/put returned an etag. But in my tests over the wire, starring a repo ( "put") doesn't have an etag in the response header, so I have no idea why 304 is present in their docs :D

Plus ConditionalJsonResources doesn't handle post/put requests.

Reply:
@criske but this is a GET, not a PUT and any get will be wored through ConditionalJsonResources, right?

Well, if this GET does not return an Etag, then it is ok.